### ISSUE TITLE
Please reconsider deleting this from pypi

### ISSUE THREAD

**Issue by beniwohli** [pylola/requirements.txt#1]
I like a good joke as much as the next guy, but I'm really not interested to have "some nasty hacks" running on my machine due to a typo.

Congrats on the idea though!

**Comment by swistakm** [comment]
I will definitely consider this in near future. Currently I would like to leave this "typo squatting experiment" to see the long term results.

Actually there was someone from Hamburg that tried to do the [research on typo squatting in PyPI
](http://svs-repo.informatik.uni-hamburg.de). I tried to contact him in order to provide some help but I got no response and since today all his packages seem to be missing from PyPI.

So I will wait with this decision until I receive some response from this person because it is possible that such package will be in fact useful for someone's research.

Despite all, this package solves some existing problem. Maybe not the one that really deserves solution, but solves it eventually. 

And what will change I will remove from it PyPI? Someone else will be able to add it anyway. Maybe this problem should be solved in other place? PyPI tried to exclude some package strings (`requirements.txt`, `rrequirements.txt`, `requirements-txt`, and `rrequirements-txt`) but it is a lousy way of dealing with that. It is impossible to create arbitrary list of every possible typo. Note that warehouse's code that may replace pypi in the future also does that but the list of exclusions is even shorter.

I think that the real place where this breaks is the `pip`. Why does it needs to accept requirements file name as an argument? Why can't we provide list of arguments as stdin redirect or using simple pipe like following:

```
pip install < requirements.txt
./compile-requirements | pip install
```

**Comment by MCGallaspy** [comment]
Hi, just want to pop in and second the "remove from pypi" sentiment. @swistakm it's true that this is relatively harmless, but it's also just plain annoying. Without this package I would have caught my typo immediately. Instead I received a cryptic exception and wasted my time tracking it down. Very inexperienced users may become confused and waste even more time.

I don't really buy your arguments which seem to amount to "someone else will do this anyway" or that somehow this is important research. Perhaps for the Hamburg folks it was research-based -- the impression I get here is that it's so you can troll people. If you're serious about this as an experiment, you're losing nearly all of your credibility by throwing a cheeky exception. Please take it down or add a genuinely helpful message.

Edit: I'm referring to [this cheeky exception](https://github.com/pylola/requirements.txt/blob/master/setup.py#L48).

**Comment by swistakm** [comment]
I think we can improve the help message. I will do that if I find some free time or someone will submit a pull request.

**Comment by aanand** [comment]
OK, I've submitted a PR in #2.

**Comment by swistakm** [comment]
Closed as part of #2. This package won't be deleted from PyPI but instead it will raise SystemExit on installation to make clear what is wrong.
